.github/workflows/scripts/translation_check.py

```
@@ -1,295 +1,274 @@
     Returns:
-        keys: A set of all valid translation keys.
+        dict[str, set[str]]: Mapping of namespace name to its flattened key set.
+    """
+    keys_by_ns: Dict[str, Set[str]] = {}
+    for ns, path in EN_LOCALES.items():
+        if not os.path.exists(path):
+            print(f"[warn] Missing locale file: {path}", file=sys.stderr)
+            keys_by_ns[ns] = set()
+            continue
+        try:
+            with open(path, "r", encoding="utf-8") as f:
```

Ruff (0.14.10)

Unnecessary mode argument

Remove mode argument

(UP015)